frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -703,18 +873,27 @@ function DataFrame({
                 sticky: false,
                 tint: true,
               },
-              rowMarkerTheme: {
-                bgCell: theme.bgHeader,
-                bgCellMedium: theme.bgHeader,
+              rowMarkers: {
+                kind: "checkbox",
+                checkboxStyle: "square",
+                theme: {
+                  bgCell: theme.bgHeader,
+                  bgCellMedium: theme.bgHeader,
+                },
               },
-              rowMarkers: "checkbox",
               rowSelectionMode: "multi",
-              rowSelect: disabled ? "none" : "multi",
+              rowSelect: "multi",
               // Support adding rows:
-              onRowAppended: disabled ? undefined : onRowAppended,
+              onRowAppended: onRowAppended,
               // Deactivate sorting, since it is not supported with dynamic editing:
               onHeaderClicked: undefined,
             })}
+          // Disable some features when the widget is disabled:
+          {...(disabled && {
+            rowSelect: "none",
+            columnSelect: "none",
+            onRowAppended: undefined,
```

Comment:
Is it possible to move this into a variable instead of relying on the fact that these properties are set last and overwrite the previous ones?
Like

```typescript
const rowSelect = disabled ? "none" : "multi";
```

and then apply `rowSelect` only once? Its a little bit tough to look at the component and keep in mind what was already set, what is overwritten etc.

Reply:
That we the previous state, I actually just refactored this out to make it a bit simpler to read

Reply:
Would you prefer to have this moved back into the other parts?

Reply:
In some of the checks it would look like this: 
```
             columnSelect: disabled
                ? "none"
                : isMultiColumnSelectionActivated
                ? "multi"
                : "single",
```

Reply:
Ok, I move it up again.